share/wake/lib/core/list.wake

```diff
@@ -23,46 +23,61 @@ package wake
 #   - the comma operator to prepend elements to a List
 #
 # Because the comma operator is right associative, a List parses like this:
+#   ```
 #   0, 1, 2, Nil = (0, (1, (2, Nil)))
+#   ```
 #
 # Since List has a type parameter, any type can be legally stored in a list
+#   ```
 #   0, 1, 2, Nil         # ok! This is a `List Integer`
 #   "0", "1", "2", Nil   # ok! This is a `List String`
+#   ```
 #
-# However, mixing types it NOT legal:
+# However, mixing types is *not* legal:
+#   ```
 #   0, "1", Nil          # bad! This will trigger a type-check failure.
+#   ```
```

Comment:
You should build and install the newest wake LSP when testing this PR. In particular, wake now adds pop-up text for `List`, `Nil`, and `,`. For the two latter constructors, it combines the comments for `List` with any specific comments for the constructor. Therefore, I think it may make sense (especially for the all important List type) to include specific documentation for the constructors, to set a good example.

Reply:
How do I install the latest one?

Reply:
Probably easiest is to grab the 'vsix' from the circle CI artifacts on the 'release' green check mark of a PR.